src/main/src/main/java/org/geoserver/data/util/IOUtils.java

```diff
@@ -54,10 +54,18 @@ private IOUtils() {
      * @throws IOException
      */
     public static void copy(InputStream from, File to) throws IOException {
-        OutputStream out = null;
-        try {
-            out = new FileOutputStream(to);
-
+        copy(from, new FileOutputStream(to));
+    }
+    
+    /**
+     * Copies the provided input stream onto an outputstream
+     * 
+     * @param from
+     * @param to
+     * @throws IOException
+     */
+    public static void copy(InputStream from, OutputStream out) throws IOException {
```

Comment:
@bencaradocdavies  The great benefit of this in combination with resources is that you can write things like:
      copy (resource.in(), other.out())
Saving yourself the declaration of two variables and reducing 5 lines to one.

The issue I ever have is that there is also a IOUtils.copy in apache-commons that has the same signature and DOESN'T close. I have replaced as many as possible of those by our own, but you have to check the imports at the top of the file to know which one is being used, which I find very risky.

To make things worse, there is also a org.geoserver.rest.util.IOUtils with a lot of duplicated code that is only being used in the rest modules. This class however has a method of slightly different signature: copy(InputStream in, OutputStream out, boolean close) where you are forced to explicitly state you are closing the stream, which I find better